source/blog/2018-10-31-fedora-atomic-28-to-29-upgrade.md

```diff
@@ -0,0 +1,175 @@
+```
+
+You can see from the command that we are adding a new remote known as
+`fedora-atomic-29` with a remote url of `https://dl.fedoraproject.org/atomic/repo/`
+We are also setting the `gpgkeypath` variable in the configuration for
+the remote. This tells OSTree that we want commit signatures to be
+verified when we download from a remote.
+
+Now that we have our `fedora-atomic-29` remote we can do the upgrade!
+
+```nohighlight
+[vagrant@host ~]$ sudo rpm-ostree rebase fedora-atomic-29:fedora/29/x86_64/atomic-host
+26 delta parts, 9 loose fetched; 264197 KiB transferred in 44 seconds        
+Upgraded:                                      
+  NetworkManager 1:1.10.12-1.fc28 -> 1:1.12.4-1.fc29        
+  NetworkManager-libnm 1:1.10.12-1.fc28 -> 1:1.12.4-1.fc29
+  NetworkManager-team 1:1.10.12-1.fc28 -> 1:1.12.4-1.fc29
+  acl 2.2.53-1.fc28 -> 2.2.53-2.fc29                
+  atomic 1.22.1-2.fc28 -> 1.22.1-27.gitb507039.fc29       
+  atomic-devmode 0.3.7-4.fc28 -> 0.3.7-6.fc29        
+  atomic-registries 1.22.1-2.fc28 -> 1.22.1-27.gitb507039.fc29
+  attr 2.4.48-3.fc28 -> 2.4.48-3.fc29              
+...
+  python-unversioned-command-2.7.15-10.fc29.noarch
+  python3-pyyaml-4.2-0.1.b4.fc29.x86_64
+Run "systemctl reboot" to start a reboot
+[vagrant@host ~]$ sudo systemctl reboot
+[vagrant@host ~]$ Connection to 192.168.121.95 closed by remote host.
+Connection to 192.168.121.95 closed.
+```
+
+After reboot we can log in and see the status:
+
+```nohighlight
+$ vagrant ssh 
+Last login: Wed Oct 31 21:14:44 2018 from 192.168.121.1
+[vagrant@host ~]$ rpm-ostree status
+State: idle
+AutomaticUpdates: disabled
+Deployments:
+● ostree://fedora-atomic-29:fedora/29/x86_64/atomic-host
+                   Version: 29.20181025.1 (2018-10-25 14:46:54)
+                    Commit: 4a999b4b303b47468ff1464051a14fd075d2e7b8bb647584b7cc80fed48cf27b
+              GPGSignature: Valid signature by 5A03B4DD8254ECA02FDA1637A20AA56B429476B4
+
+  ostree://fedora-atomic:fedora/28/x86_64/atomic-host
+                   Version: 28.20181007.0 (2018-10-07 20:43:44)
+                    Commit: 8df48fa2e70ad1952153ae00edbba08ed18b53c3d4095a22985d1085f5203ac6
+              GPGSignature: Valid signature by 128CF232A9371991C8A65695E08E7E629DB62FB1
+[vagrant@host ~]$
+[vagrant@host ~]$ cat /etc/os-release | head -n 2
+NAME=Fedora
+VERSION="29.20181025.1 (Atomic Host)"
+```
+
+We are now on Fedora 29 Atomic Host. Now is a good time to check your
+services (most likely running in containers) to see if they are still
+working. If not, then you always have the rollback command: `sudo
+rpm-ostree rollback`.
+
+**NOTE:** Over time you can see updated commands for upgrading Atomic
+          Host between releases by visiting [this](https://fedoraproject.org/wiki/Atomic_Host_upgrade)
+          wiki page.
+
+# Appendix A: Using older versions of Kubernetes/OpenShift on Fedora 29 Atomic Host
+
+XXX jbrooks fill in
```

Comment:
@jasonbrooks - can you fill in this part?

Reply:
Due to [an issue](https://pagure.io/atomic-wg/issue/512) that arises from the combination of the version of systemd that ships with Fedora 29 Atomic Host, and the version of runc that's bundled with Kubernetes and with OpenShift Origin, users of certain Kubernetes and OpenShift Origin releases must take steps to keep their Kubernetes or Origin clusters running following an upgrade to 29.

Users running Kubernetes releases older than v1.10.4 or OpenShift Origin releases older than v3.11 would do best to upgrade their Kubernetes or Origin clusters to a more recent version before undertaking an upgrade to Fedora 29 Atomic Host.

For users who prefer to upgrade to 29 without changing their cluster version, it's possible to work around the issue by changing the `croupdriver` for both docker and for either the kubelet or the orgin-node component from its default value of `systemd` to `cgroupfs`.

## For docker (applies to both Kubernetes and OpenShift Origin clusters):

```
# cp /usr/lib/systemd/system/docker.service /etc/systemd/system/

# sed -i 's/cgroupdriver=systemd/cgroupdriver=cgroupfs/' /etc/systemd/system/docker.service
```

## For kubelet (applies to Kubeadm-based Kubernetes clusters):

```
# sed -i 's/cgroup-driver=systemd/cgroup-driver=cgroupfs/' /etc/systemd/system/kubelet.service.d/kubeadm.conf
```

## For origin-node (applies to OpenShift Origin clusters):

```
# vi /etc/origin/node/node-config.yaml
```

Add the following element to the `kubeletArguments:` section:

```
cgroup-driver:
  - "cgroupfs"
```

After making these changes, rebase to 29 as directed above, reboot, and the nodes should return in `Ready` state as expected.

Reply:
@jasonbrooks updated!